### PR TITLE
wgsl: Change *Unbounded[Interval|Vector|...] to *Infinite[Interval|Vector|...]

### PR DESCRIPTION
**Pull request by zoddicus**
The current 'unbounded' constants are for [-∞, ∞] intervals. Changing
the name to avoid confusion once the 'bounded' state bit is added to
the interval class.

Issue  #3201

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by zoddicus** [comment]
This is dependent on the `bounds` -> `endpoints` change, so new code is in the second diff, bd95de97a2a25296922b3e0d6331fd6c9f727cc6

File: src/webgpu/util/floating_point.ts [hunk 1]
```diff
@@ -4259,7 +4259,7 @@ export abstract class FPTraits {
    * [0, 0] and [1, 1] indicate that the correct answer in point they encapsulate.
    * [0, 1] should not be treated as a span, i.e. 0.1 is acceptable, but instead
    * indicate either 0.0 or 1.0 are acceptable answers.
-   * [-∞, +∞] is treated as unbounded interval, since an unbounded or
+   * [-∞, +∞] is treated as unbounded interval, since an out of bounds or
    * infinite value was passed in.
```
Comment (jzm-intel) [inline src/webgpu/util/floating_point.ts:4263]: How about
``` suggestion
   * [-∞, +∞], i.e. infinite interval, will be returned when an out-of-bound or
   * infinite value was passed in.
```

**Review by jzm-intel — APPROVED** [review]
The code change generally looks good to me, thanks!

But I am not sure about the usage of name `infinite`. If I get it right, we use "unbounded interval" to indicate that we accept any result of the given FP type, or as the WGSL spec said "indeterminate value". But technically yes, it is a interval of [-inf, inf].

I hope others can have comments on this.

**Comment by jzm-intel** [comment]
> The code change generally looks good to me, thanks!
> 
> But I am not sure about the usage of name `infinite`. If I get it right, we use "unbounded interval" to indicate that we accept any result of the given FP type, or as the WGSL spec said "indeterminate value". But technically yes, it is a interval of [-inf, inf].
> 
> I hope others can have comments on this.

@ben-clayton @dneto0

**Comment by zoddicus** [comment]
This is the issue that I have filed, https://github.com/gpuweb/cts/issues/3139

In brief, for mix specifically, I have written a test case that goes OOB in one of the intermediate steps, but comes back in bounds by the end. This leads to a situation where (because of the intermediate OOB) the CTS should accept any input, but because we have an in bounds end interval, it is checking against that.

As implemented the framework is supposed to have converted that OOB value to the infinite interval at the intermediate step (which I probably could find the bug and fix) but that would eliminate the endpoint values. For validation there is distinction between going OOB near the edge of the range versus being way out there, so retaining what the actual calculated endpoints are is going to be needed for the validation testing.

**Comment by zoddicus** [comment]
The bit should resolve this issue by allowing .contains to be rewritten so that if bit == true return true at the beginning.

**Comment by ben-clayton** [comment]
> In brief, for mix specifically, I have written a test case that goes OOB in one of the intermediate steps, but comes back in bounds by the end. This leads to a situation where (because of the intermediate OOB) the CTS should accept any input, but because we have an in bounds end interval, it is checking against that.

This is roughly what I understood, but I'm vague about:
(a) How it goes OOB and then back into bounds.
(b) Why the modeling of this out-then-in-bounds doesn't accurately replicate the desired outcome.

Would it be possible to describe this in steps, with a concrete example, highlighting where the mismatch occurs?

**Comment by zoddicus** [comment]
Took me a moment to come up with a solid example, because my new workstation apparently does the math in a higher precision, so wasn't causing the CTS to trip anymore :upside_down_face: 

For mix one of the allowed formulas is `e1 * (1 - e3) + e2 * e3`
If we use these inputs
```
e1 = 1                                                                                                                                           
e2 = 2                                                                                                                                           
e3 = (max/2) + 1
```

We have the following operations:
1)
```
e2 * e3 = 2 * (max/2) + 2                                                                                                                        
        = max + 2
```

2)
```
1 - e3 = 1 - max/2 + 1                                                                                                                           
       = -max / 2 
```

3)
```
e1 * (1 - e3) = 1 * ( -max/2)                                                                                                                    
              = -max/2
```

4)
```
e1 * (1 - e3) + e2 * e3 = -max/2 + (max + 2)                                                                                                     
                        = max/2 + 2
```

So the final result 4) is in bounds, but 1) went OOB, and if we are operating in f16, number in TS can accurately represent all of this.

The framework for f16 currently returns [32752.0, 32768.0], which is in bounds. The current implementation isn't behaving as intended, because at 1) this should have be splatted to [-∞, ∞], which should have propegated through. This in theory could be fixed, but will need to be revisited for validation testing.

Specifically https://www.w3.org/TR/WGSL/#floating-point-overflow states that if a value is OOB, but near the edge of the range, than it is fair for a platform to round into range, so doesn't nescarily cause a validation error. So how far things go OOB needs to be tracked, so this unbounded bit will probably need to be expanded to be a tri-state, always in bounds, always alteast near OOB, or went far OOB.

**Pull request closed without merging**